wordpress-mobile/WordPress-Android
# WordPress Stories : Story publish bottom sheet integration part 2

Fixes #12305

## Builds on Top of https://github.com/wordpress-mobile/WordPress-Android/pull/12306
## Solution
To perform this integration with the Story Composer, these are the main things that were done. 

- The bottom sheet was supplied with a boolean param that determines if the `Story Title UI` should be shown. 
- The `StoryComposerActivity ` implemented the `EditPostActivityHook` that's necessary for the functionality of the bottom sheet to update the post with the changes that are taking place on the visibility, publish date and tags. 
- `PrepublishingEventProvider` a component that allows the host app to respond to the next button event, that allows the bottom sheet to be shown before the publish operation is completed. 
- `EditPostRespository` now responds to all changes made in the bottom sheet and persists those changes to the database so if there is a configuration change the user doesn't lose the changes they made. 
- Updating the story title causes the current title to be updated. 

## Testing
1. Create a new story post. 
2. Click Publish. 
3. See the bottom sheet like below. 
4. Add a story title.

Before | After 
--------|-------
<kbd><img src="https://user-images.githubusercontent.com/1509205/85955904-8695c400-b947-11ea-80f0-617458e97140.png" width="320"></kbd>   |     <kbd><img src="https://user-images.githubusercontent.com/1509205/85955877-577f5280-b947-11ea-80c5-d575746f6061.png" width="320"></kbd>
  
## Concerns
1. Currently, the bottom sheet isn't above the keyboard due to the Story Composer being a full-screen activity. A custom solution will be created to solve this issue.
<kbd><img src="https://user-images.githubusercontent.com/1509205/85955920-a62cec80-b947-11ea-8922-b5b08289b3ef.png" width="320"></kbd>

2. The button in the bottom sheet says "Update Now" instead of "Publish Now". This is because the `PostStatus` is `Published`. Based on current design, once the post is new it should say "Draft" so that the current button is "Publish Now". The current post publishing flow in the `EditPostActivity` sets the `PostStatus` to "Published" when necessary before the post is uploaded, so this will be investigated to ensure that it's done appropriately either by that flow or by our own logic, where once the post is set to be uploaded we can then mark it as being "Published."

<kbd><img src="https://user-images.githubusercontent.com/1509205/85956020-24898e80-b948-11ea-8442-9172f92a6d39.png" width="320"></kbd>

3. The bottom sheet has other operations i.e it allows the user to change the status to others that wouldn't publish the post now but schedule it, mark as review, etc. We will need to investigate this and ensure that the user is able to save a draft story post, or schedule one. 

Visibility Options | Submit Button Changed 
--------|-------
 <kbd><img src="https://user-images.githubusercontent.com/1509205/85956031-3ec36c80-b948-11ea-8468-5227e20b2a18.png" width="320"></kbd>  |       <kbd><img src="https://user-images.githubusercontent.com/1509205/85956035-4551e400-b948-11ea-8a64-5835a8e7eb77.png" width="320"></kbd>

4. This PR didn't implement any dismissal logic that would hide buttons etc. so that will be done in another PR. 

5. The flow for setting the current title to the StoryRepository needs verification and testing. Also should the Story Title be set as the post title? 

## Reviewing 
Only 1 reviewer is needed but anyone can review. 

## Submitter Checklist
